keps/sig-node/3857-rro-mounts/README.md

```diff
@@ -0,0 +1,1075 @@
++	//   2.2. If RecursiveReadOnly is set to Disabled, the mount is not made recursively read-only.
++	//   2.3. If RecursiveReadOnly is set to IfPossible, the mount is made recursively read-only,
++	//        if it is supported by the runtime.
++	//   2.4. If RecursiveReadOnly is set to Enabled, the mount is made recursively read-only.
++	//        MountPropagation must be None or nil (which defaults to None).
++	//
++	// +featureGate=RecursiveReadOnlyMounts
++	// +optional
++	RecursiveReadOnly *RecursiveReadOnlyMode
+ 	// Required. If the path is not an absolute path (e.g. some/path) it
+ 	// will be prepended with the appropriate root prefix for the operating
+ 	// system.  On Linux this is '/', on Windows this is 'C:\'.
+@@ -1870,6 +1887,7 @@ type VolumeMount struct {
+ 	// to container and the other way around.
+ 	// When not set, MountPropagationNone is used.
+ 	// This field is beta in 1.10.
++	// When RecursiveReadOnly is set to Enabled, MountPropagation must be None or nil (which defaults to None).
+ 	// +optional
+ 	MountPropagation *MountPropagationMode
+ 	// Expanded path within the volume from which the container's volume should be mounted.
+@@ -1905,6 +1923,18 @@ const (
+ 	MountPropagationBidirectional MountPropagationMode = "Bidirectional"
+ )
+ 
++// RecursiveReadOnlyMode describes recursive-readonly mode.
++type RecursiveReadOnlyMode string
++
++const (
++	// RecursiveReadOnlyDisabled disables recursive-readonly mode.
++	RecursiveReadOnlyDisabled RecursiveReadOnlyMode = "Disabled"
++	// RecursiveReadOnlyIfPossible enables recursive-readonly mode if possible.
++	RecursiveReadOnlyIfPossible RecursiveReadOnlyMode = "IfPossible"
++	// RecursiveReadOnlyEnabled enables recursive-readonly mode, or raise an error.
++	RecursiveReadOnlyEnabled RecursiveReadOnlyMode = "Enabled"
++)
++
+ // VolumeDevice describes a mapping of a raw block device within a container.
+ type VolumeDevice struct {
+ 	// name must match the name of a persistentVolumeClaim in the pod
+@@ -2505,6 +2535,10 @@ type ContainerStatus struct {
+ 	// +featureGate=InPlacePodVerticalScaling
+ 	// +optional
+ 	Resources *ResourceRequirements
++	// Status of volume mounts.
++	// +listType=atomic
++	// +optional
++	VolumeMounts []VolumeMountStatus
+ }
+ 
+ // PodPhase is a label for the condition of a pod at the current time.
+@@ -2584,6 +2618,24 @@ const (
+ 	PodResizeStatusInfeasible PodResizeStatus = "Infeasible"
+ )
+ 
++// VolumeMountStatus shows status of volume mounts.
++type VolumeMountStatus struct {
++	// Corresponds to the name of the original VolumeMount.
++	Name string
++	// Corresponds to the mount path of the original VolumeMount.
++	MountPath string
++	// ReadOnly must be set to true or false.
++	// A nil value in the original VolumeMount must be translated to true or false,
++	// depending on the mount result.
++	ReadOnly bool
++	// RecursiveReadOnly must be set to Disabled, Enabled, or nil (for non-readonly mounts).
```

Comment:
I'd like to argue about the more, but we can do that in actual API review.  For the purposes of the KEP, this is clear enough.